### PR TITLE
Kaizen/#178/self process monitoring

### LINKED ISSUE #178: feature request - monitor and re-lauch dead process
Would be very nice to have active monitoring on all processes and if some of the services dies to relaunch it automatically. 
[Monit] (https://mmonit.com/monit/) is a lightweight and does very good job - active monitor and signaling. 

Today I managed to kill shadosocks three times. I had every time to ssh to start it again. 

Keep up the great work!

### PR DESCRIPTION
**Pull request by Russell-IO**
Adds monitoring & automatic process relaunching for Nginx / Stunnel / OpenVPN / OpenSSH / dnsmasq / TinyProxy / SSLH :)
Resolves #178

### CODE REVIEW

**Comment by Russell-IO** [comment]
This also adds a /monit url to the gateway where you can view service status etc (requires same auth as main portal..)

**Comment by mveplus** [comment]
That's a great job! 
Would be great if to-be added a few missing services to look after too: shadowsocks, strongswan, xl2tpd, tor, and obfsproxy

File: playbooks/roles/monit/templates/monitrc.j2 [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+# Tiny Proxy
+check process tinyproxy with pidfile /var/run/tinyproxy/tinyproxy.pid
+   start program  "/etc/init.d/tinyproxy restart"
+   stop program  "/etc/init.d/tinyproxy stop"
+   if failed host 127.0.0.1 port 8888 type tcp then restart
+   # if 5 restarts within 5 cycles then timeout
+
+# SSLH
+check process tinyproxy with pidfile /var/run/sslh/sslh.pid
```
Comment (jlund) [inline playbooks/roles/monit/templates/monitrc.j2]: s/tinyproxy/sslh/

File: playbooks/streisand.yml [hunk 2]
```diff
@@ -14,5 +14,6 @@
     - tinyproxy
     - tor-bridge
     - sslh
+    - monit
```
Comment (jlund) [inline playbooks/streisand.yml]: We will want to add this to the Amazon and GCE playbooks as well.

File: playbooks/roles/monit/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+---
+- name: Install Monit
+  apt: name=monit
+
+- name: Update Monit configuration
+  template: src=monitrc.j2
+        dest=/etc/monit/monitrc
```
Comment (jlund) [inline playbooks/roles/monit/tasks/main.yml]: These lines should line up with the `src` line.

**Comment by jlund** [comment]
Thanks for this. Sorry that it took me so long to review. I found a couple of little minor things to clean up. I'm excited to get this merged.

File: playbooks/streisand.yml (hunk 2 above)
Comment (Russell-IO) [inline playbooks/streisand.yml]: done ;)

File: playbooks/roles/monit/tasks/main.yml (hunk 3 above)
Comment (Russell-IO) [inline playbooks/roles/monit/tasks/main.yml]: done

File: playbooks/roles/monit/templates/monitrc.j2 (hunk 1 above)
Comment (Russell-IO) [inline playbooks/roles/monit/templates/monitrc.j2]: @jlund - don't understand this comment

**Comment by Russell-IO** [comment]
Think i've cleaned everything @jlund - let me know ?

File: playbooks/roles/monit/templates/monitrc.j2 (hunk 1 above)
Comment (Russell-IO) [inline playbooks/roles/monit/templates/monitrc.j2]: Ahhh Just did a clean run though and it all makes sense :)...  push on the way

**Comment by Russell-IO** [comment]
@jlund found the tinyproxy bug ;) should be golden now.

**Comment by jlund** [comment]
I should be able to take a look tonight or tomorrow. I really appreciate your help.

**Comment by Russell-IO** [comment]
No problem at all :) Happy to be a contributor.

**Comment by jlund** [comment]
Merged! I had to make a few minor adjustments that were likely necessary because of the recent switch from Debian 7 to Ubuntu 14.04. Thanks again.

**Pull request merged**